project/src/main/java/com/google/sps/servlets/NewsService.java

```diff
@@ -0,0 +1,47 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import okhttp3.*;
+
+import java.io.IOException;
+
+/** Service class that delivers news under a requested topic  */
+public class NewsService {
+  // one instance, reuse
+  private final OkHttpClient httpClient = new OkHttpClient();
+
+  public String retrieveNewArticles(String topic) {
```

Comment:
Let's add a comment explaining this method
Are there restrictions on the topic? Can it have spaces, can it be null, what about non A-Z characters?